cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -112,6 +133,101 @@ void kafka_consumer::consume_to_buffer()
   }
 }
 
+std::map<std::string, std::string> kafka_consumer::current_configs()
+{
+  std::map<std::string, std::string> configs;
+  std::list<std::string> *dump = kafka_conf->dump();
+  std::string key;
+  std::string val;
+  for (std::list<std::string>::iterator it = dump->begin(); it != dump->end();) {
+    key = (*it);
+    it++;
+    val = (*it);
+    it++;
+    configs.insert(std::pair<std::string, std::string>{key, val});
+  }
+  return configs;
+}
```

Comment:
This can be simplified a lot. @codereport and I brainstormed this together
https://godbolt.org/z/TMc39P

```suggestion
std::map<std::string, std::string> kafka_consumer::current_configs()
{
  std::map<std::string, std::string> configs;
  std::list<std::string> *dump = kafka_conf->dump();
  for (auto it = dump->begin(); it != dump->end(); std::advance(it, 2))
    configs.insert({*it, *std::next(it)});
  return configs;
}
```